provisioning/src/main/java/org/wildfly/build/util/POSIXFinePrivilegesHelper.java

```diff
@@ -0,0 +1,46 @@
+package org.wildfly.build.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+
+import static org.wildfly.build.provisioning.ServerProvisioner.OS_WINDOWS;
+
+public class POSIXFinePrivilegesHelper {
+
+    // TODO: ugly
+    private static final Path NO_PATH_PRODIVED = null;
+
+    private static final String JAR_FILE_MASK = "rw-r--r--";
+    private static final String PROPERTIES_FILE_MASK = "rw-------";
+
+    public static void fineTunePrivileges(Path path) throws IOException {
+        if (!OS_WINDOWS) Files.setPosixFilePermissions(path,getPermissionsMasks(NO_PATH_PRODIVED,path));
+    }
+
+    private static Set<PosixFilePermission> getPermissionsMasks(Path src, Path dest) {
+        final String suffix = dest.toAbsolutePath().toString();
+        if ( suffix.endsWith(".jar") ) {
+            System.out.println("JAR:" + dest);
+            return PosixFilePermissions.fromString(JAR_FILE_MASK);
+        } else if ( suffix.endsWith(".properties") ) {
+            System.out.println("PROPS:" + dest);
+            return PosixFilePermissions.fromString(PROPERTIES_FILE_MASK);
+        } else
+            return getDefaultPermissionsFor(src,dest);
+    }
+
+    private static Set<PosixFilePermission> getDefaultPermissionsFor(Path src, Path dest) {
+        try {
+            if ( src == NO_PATH_PRODIVED ) //FIXME: it's not really elegant to say the least
```

Comment:
why not just use null check here?

Reply:
for lisibility purpose, but I hope to rework the class to remove this.

Reply:
In this case the `src` is always going to be `null` anyway. The `src` parameter should just be removed on both this method and the `getPermissionsMasks()`.